src/spartan_router.erl

```diff
@@ -13,30 +13,24 @@
 -spec(upstreams_from_questions(dns:questions()) -> ordsets:ordset(upstream())).
 upstreams_from_questions([#dns_query{name=Name}]) ->
     Labels = spartan_app:parse_upstream_name(Name),
-    lists:map(fun normalize_ip/1, find_upstream(Name, Labels));
```

Comment:
I'll pull the checking into a separate function. The reason i removed it is because i think that it's a flawed design pattern to accept all kinds of random types and then coerce them all into a single one. I find it preferable for functions to return the correct type in the first place.